packages/core/contracts/UnirepSocial.sol

```diff
@@ -265,4 +269,39 @@ contract UnirepSocial {
     ) external {
         unirep.updateUserStateRoot(publicSignals, proof);
     }
+
+     /*
+     * Set new user name for an epochKey
+     * @param epochKey epoch key that attempts to set a new uername
+     * @param oldUsername
+     * @param newUsername requested new user name
+     */
+     function setUsername(
+        uint256 epochKey,
+        uint256 oldUsername,
+        uint256 newUsername
+     ) external payable {
+        (,,,,,,,uint attestingFee,,) = unirep.config();
```

Comment:
We have more usable getter functions for these values now. This can be replaced with `uint attestingFee = unirep.attestingFee()`.